lib/spack/spack/fetch_strategy.py

```diff
@@ -392,7 +392,26 @@ def __init__(self, *args, **kwargs):
 
     @_needs_stage
     def fetch(self):
-        super(CacheURLFetchStrategy, self).fetch()
+        path = re.sub('^file://', '', self.url)
+
+        # directly check whether the cache file already exists.
+        if not os.path.isfile(path):
+            raise NoCacheError('No cache of %s' % path)
+
+        # Just symlink it in if so.
+        self.stage.chdir()
+        basename = os.path.basename(path)
```

Comment:
This will choose the stage filename according to the cache path. This may cause issues if the download is a single file (vs. an archive) and the package logic expects to work with just the filename. I'd prefer using self.stage.save_filename instead.